src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -756,6 +756,17 @@ JvmtiEnvBase::get_thread_state(oop thread_oop, JavaThread* jt) {
   return state;
 }
 
+jint
+JvmtiEnvBase::get_thread_state(oop thread_oop, JavaThread* jt) {
+  jint state = get_thread_state_base(thread_oop, jt);
+
+  if (is_passive_carrier_thread(jt, thread_oop)) {
+    state = JVMTI_THREAD_STATE_ALIVE | JVMTI_THREAD_STATE_WAITING |
+            JVMTI_THREAD_STATE_WAITING_INDEFINITELY;
+  }
+  return state;
+}
```

Comment:
You don't need to call get_thread_state_base in case "passive carrier thread":

    if (is_passive_carrier_thread(jt, thread_oop)) {
      return JVMTI_THREAD_STATE_ALIVE | JVMTI_THREAD_STATE_WAITING
             | JVMTI_THREAD_STATE_WAITING_INDEFINITELY;
    }
    return get_thread_state_base(thread_oop, jt);

Reply:
Thanks. Yes, noticed it. :) Fixed now.